pfr/engine/etl_eventlog.py

```diff
@@ -55,35 +50,63 @@ def _get_case_edge(self, sublog):
         eventlog = pd.DataFrame()
+        if self._config['eventlogInputColumns']['caseId'] not in raw_eventlog.columns:
+            print(f'The case id column set in configuration file is not present in passed eventlog file')
+            exit(1)
+        if self._config['eventlogInputColumns']['activity'] not in raw_eventlog.columns:
+            print(f'The activity column set in configuration file is not present in passed eventlog file')
+            exit(1)
         eventlog[Eventlog.case_id.name] = raw_eventlog[self._config['eventlogInputColumns']['caseId']].astype(str)
-
-
         eventlog[Eventlog.activity.name] = raw_eventlog[self._config['eventlogInputColumns']['activity']]
         eventlog[Eventlog.activity.name] = eventlog.apply(lambda row: trim_if_string(row[Eventlog.activity.name]), axis=1)
         eventlog[Eventlog.id.name] = range(0, len(eventlog))
-        eventlog[Eventlog.activity_end_ts.name] = raw_eventlog[self._config['eventlogInputColumns']['endTimestamp']]
-        eventlog[Eventlog.activity_end_ts.name] = pd.to_datetime(eventlog[Eventlog.activity_end_ts.name],
-                                                            errors='raise',
-                                                            format=timestamp_mask,
-                                                            utc=True).dt.tz_localize(None)
 
-        if "resource" in self._config['eventlogInputColumns'].keys():
-            eventlog[Eventlog.resource.name] = raw_eventlog[self._config['eventlogInputColumns']['resource']]
+        if (self._config['eventlogInputColumns'].get('endTimestamp') not in raw_eventlog.columns
+                and self._config['eventlogInputColumns'].get('startTimestamp') not in raw_eventlog.columns):
+            print(f'The timestamp columns set in configuration file is not present in passed eventlog file')
+            exit(1)
+
+        if self._config['eventlogInputColumns'].get('endTimestamp'):
+            eventlog[Eventlog.activity_end_ts.name] = raw_eventlog[self._config['eventlogInputColumns']['endTimestamp']]
         else:
-            eventlog[Eventlog.resource.name] = ""
+            eventlog[Eventlog.activity_end_ts.name] = raw_eventlog[self._config['eventlogInputColumns']['startTimestamp']]
+
+        if self._config['eventlogInputColumns'].get('startTimestamp'):
+            eventlog[Eventlog.activity_start_ts.name] = raw_eventlog[self._config['eventlogInputColumns']['startTimestamp']]
+        else:
+            eventlog[Eventlog.activity_start_ts.name] = raw_eventlog[self._config['eventlogInputColumns']['endTimestamp']]
 
-        if "startTimestamp" in self._config['eventlogInputColumns'].keys():
+        try:
+            eventlog[Eventlog.activity_end_ts.name] = pd.to_datetime(
+                eventlog[Eventlog.activity_end_ts.name],
+                errors='raise',
+                format=timestamp_mask,
+                utc=True).dt.tz_localize(None)
+        except ValueError:
```

Comment:
Jeśli chcesz złapać konkretny wyjątek to robisz
```
except ValueError as ex:
    if ex cośtam:
        print("Bład taki a taki:", ex)
```

Reply:
done